ui/src/reducers/beamline.js

```diff
@@ -150,8 +150,7 @@ export default (state = INITIAL_STATE, action) => {
 
     case 'BL_UPDATE_HARDWARE_OBJECT': {
       const attrData = Object.assign(
-        state.hardwareObjects[action.data.name] || {},
-        action.data
+        {}, {...state.hardwareObjects[action.data.name]}, action.data
```

Comment:
This one is interesting :), took me sometime to find

Reply:
Since you're now passing an empty object as first argument to `Object.assign`, you don't need to make a copy of the second argument — it will not be mutated.